src/score_hv/harvesters/inc_logs.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python
+
+""" Collection of methods to faciliate retrieval of increment
+    descriptive statistics from logs
+"""
+
+import os
+
+from collections import namedtuple
+from dataclasses import dataclass
+from dataclasses import field
+
+from score_hv.config_base import ConfigInterface
+
+HRVSTER_NAME = 'inc_logs'
+VALID_STATISTICS = ('mean', 'RMS')
+VALID_VARIABLES = ['pt_inc', 's_inc', 'u_inc', 'v_inc', 'SSH', 'Salinity',
+                   'Temperature', 'Speed of Currents', 'o3mr_inc', 'sphum_inc',
+                   'T_inc', 'delp_inc', 'delz_inc']
+
+HarvestedData = namedtuple('HarvestedData', ['logfile',
+                                             'statistic',
+                                             'variable',
+                                             'value', 
+                                             'units'])
+N_TUPLE_ENTRIES = 5                                             
+
+@dataclass
+class LogIncCfg(ConfigInterface):
+    """ Dataclass to hold and provide configuration information pertaining to
+        how the harvester should read increment log files
+    
+        Parameters:
+        -----------
+        config_data: dict - contains configuration data parsed from either an
+                            input yaml file or input dict
+        
+    """
+    config_data: dict = field(default_factory=dict)
+    
+    def __post_init__(self):
+        self.set_config()
+    
+    def set_config(self):
+        """ function to set configuration variables from given dictionary
+        """ 
+        try:
+            self.harvest_filename = self.config_data.get('filename')
+        except Exception as err:
+            msg = '%s key missing' % key
+            raise KeyError(msg) from err
+        
+        self.set_stats()
+        self.set_variables()
+    
+    def set_variables(self):
+        """
+        set the variables specified by the config dict
+        """
+        try:
+            self.variables = self.config_data.get('variable')
+            for var in self.variables:
+                if var not in VALID_VARIABLES:
+                    msg = f'Invalid variable, must be one of \'{VALID_VARIABLES}\''
+                    raise KeyError(msg)
+        except Exception as err:
+            msg = f'\'variables\' key missing, must be one of ' \
+                f'({VALID_VARIABLES}) - err: {err}'
+            raise KeyError(msg) from err
+
+    def set_stats(self):
+        """
+        set the statistics specified by the config dict
+        """
+        try:
+            self.stats = self.config_data.get('statistic')
+            for stat in self.stats:
+                if stat not in VALID_STATISTICS:
+                    msg = f'Invalid statistic, must be one of \'{VALID_STATISTICS}\''
+                    raise KeyError(msg)
+        except Exception as err:
+            msg = f'\'statistics\' key missing, must be one of ' \
```

Comment:
same comment as with variables, the key name in the error needs to match whats being called in the code

Reply:
removed the try/except block and corrected the message